pkg/queue/queue.go

```diff
@@ -9,21 +9,23 @@ import (
 	"time"
 )
 
-func New(initialConcurrency int) *Queue {
+func New(initialConcurrency int, concurrencyPerReplica uint32) *Queue {
 	return &Queue{
-		concurrency: initialConcurrency,
-		list:        list.New(),
-		enqueued:    make(chan struct{}),
-		completed:   make(chan struct{}),
+		concurrency:           initialConcurrency,
+		concurrencyPerReplica: concurrencyPerReplica,
+		list:                  list.New(),
+		enqueued:              make(chan struct{}),
+		completed:             make(chan struct{}),
 	}
 }
 
 // Queue is a thread-safe FIFO queue that will limit the number of concurrent
 // requests that can be in progress at any given time.
 type Queue struct {
-	// concurrency is the max number of in progress items.
-	concurrency    int
 	concurrencyMtx sync.RWMutex
+	// concurrency is the max number of in progress items.
+	concurrency           int
+	concurrencyPerReplica uint32
```

Comment:
I would prefer to use `int` everywhere to keep it the same. No strong preference though.

Reply:
or maybe we should change to `uint` everywhere? Whatever we do, ideally we do it consistently unless there is a strong reason not to.

Reply:
+1 for consistency. I will use `int` here and we change it in a new PR